### DISCUSSION TITLE
mitmproxy as a blocker - how to add multiple elements

### DISCUSSION THREAD

**Discussion by SebastianMpl** [mitmproxy/mitmproxy#6751] [Q&A] [answered]
Hi Guys,

I wanted to experiment with blocking  various things.
Based on the text:
https://dev.to/dandyvica/use-mitmproxy-as-a-personal-firewall-4m6h
I configured mitmproxy and created those two files.
The file with url links has...well url links:
```
scorecardresearch\.com
geo\.yahoo\.com
adversting\.
ads\.
\.taboola\.
\.doubleclick\.
\.xiti\.
criteo
\.pubmatic\.
\.chartbeat\.net
\.twimg\.com
\.keywee\.co
\.\w+adserver\.com
outbrain
\.bouncex\.
\.facebook\.
\.google\.com
banners
banner
sensorCollect
```

I still need to learn the patterns but what I can see is that the blocking based on url (or its part) works very well.

What should I do to block an element based on e.q css id or class element( e.g Promoted posts on LinkedIn)
Could you please write a small example that could be added to the url file above?

Thank you,

**Comment by Prinzhorn** [comment, marked as answer]
> What should I do to block an element based on e.q css id or class element( e.g Promoted posts on LinkedIn)

This is an entirely different concept from blocking an entire URL. You'd need to parse/manipulate the HTML, which won't work for dynamically rendered stuff (React, Svelte, etc.)

This example addon manipulates the HTML, that should be enough to get you started https://github.com/mitmproxy/mitmproxy/blob/d4200a7c0d2f4efd77c44651645b59662a29a54a/examples/addons/internet-in-mirror.py

Anything beyond that point is a huge undertaking.

**Reply by SebastianMpl** [reply]
@Prinzhorn 
Thank you for your explanation and link to Python script.
I see your point and will play with it further.
